import/popups/PopupProvider.qml

```diff
@@ -0,0 +1,11 @@
+import QtQml 2.0
+
+QtObject {
+    property var contextMenu: ({"type": "item", "component": "ContextMenu.qml" })
+    property var alertPopup: ({"type": "dialog", "component": "AlertDialog.qml" })
+    property var confirmPopup: ({"type": "dialog", "component": "ConfirmDialog.qml" })
+    property var promptPopup: ({"type": "dialog", "component": "PromptDialog.qml" })
+    property var authPopup: ({"type": "dialog", "component": "AuthDialog.qml" })
+    property var passwordManagerPopup: ({"type": "dialog", "component": "PasswordManagerDialog.qml" })
+    property var locationPermissionPopup: ({"type": "dialog", "component": "LocationDialog.qml" })
```

Comment:
Why not as QML type. Works for me.

Reply:
Do you mean, why "var" ?  Because we accept either url string (allows lazy loading / compilation of the component), descriptor (allows specifying dialog type), or component instance, so the type has to accept each of those.